buildspec.yml

```diff
@@ -9,26 +9,26 @@ phases:
       - $(aws ecr get-login --no-include-email --region $AWS_DEFAULT_REGION)
       - echo Building the Docker image...
       - chmod +x docker-entrypoint.sh
-      - docker build --build-arg RAILS_ENV=$RAILS_ENV -t $IMAGE_REPO_NAME:test .
+      - docker build --build-arg RAILS_ENV=$INFRA_ENV -t $CONTAINER_NAME:test .
   build:
     commands:
-      - IMAGE_TAG=commit-$CODEBUILD_RESOLVED_SOURCE_VERSION
-      - REPOSITORY_URI=$AWS_ACCOUNT_ID.dkr.ecr.$AWS_DEFAULT_REGION.amazonaws.com/$IMAGE_REPO_NAME
       - echo Testing the newly built Docker image...
       - docker run --name test -d -e RAILS_ENV=test --env-file=docker-compose.env.example $IMAGE_REPO_NAME:test /bin/bash -c "tail -f /dev/null"
       - docker exec test bundle install --with test --retry 3 --jobs 20
       - docker rm -f test
+  post_build:
+    commands:
+      - IMAGE_TAG=commit-$CODEBUILD_RESOLVED_SOURCE_VERSION
+      - REPOSITORY_URI=$AWS_ACCOUNT_ID.dkr.ecr.$AWS_DEFAULT_REGION.amazonaws.com/$CONTAINER_NAME
+      - IMAGE_TAG=commit-$CODEBUILD_RESOLVED_SOURCE_VERSION
```

Comment:
Why does this IMAGE_TAG needs to be declared twice?

Reply:
Ah, didn't notice that, I'll remove one of them